### DISCUSSION TITLE
How do we export the entire schema to CSV using pyhton?

### DISCUSSION THREAD

**Discussion by sameer-metron** [ocsf/ocsf-schema#472] [Q&A]
How can we export the schema to a CSV file having the columns as ['Name', 'Caption', 'Description', 'Type']?

**Comment by jp-harvey** [comment]
@sameer-metron there are a couple of ways this could be done depending on the desired result and how much you want to rely on the Python code as opposed to using logic already implemented in the `ocsf-server`. For a simple use case as stated, assuming you just want a list of all the classes and objects, you could just iterate through the events and objects folders / subfolders and populate the fields in your choice of data structure (string, dict, pandas dataframe etc.) and then save to file as CSV.

Alternately the code could query the OCSF server api endpoints - especially useful if if a more complex use case was required. This would save having to construct the schema with the Python code, but not really necessary if none of the attributes and relationships are of interest.

**Comment by JasonKeirstead** [comment]
This will give it to you in JSON

curl -X GET "https://schema.ocsf.io/export/schema" -H "accept: application/json"

Converting that to CSV in a useful way depends a lot on what you are actually trying to do with the CSV, but should be pretty simple using various JSON command-line tools and/or python
